devicetypes/smartthings/zigbee-battery-accessory-dimmer.src/zigbee-battery-accessory-dimmer.groovy

```diff
@@ -0,0 +1,129 @@
+				attributeState "turningOff", label:'${name}', action:"switch.on", icon:"st.switches.light.off", backgroundColor:"#ffffff", nextState:"turningOn"
+			}
+			tileAttribute ("device.level", key: "SLIDER_CONTROL") {
+				attributeState "level", action:"switch level.setLevel"
+			}
+			tileAttribute ("device.battery", key: "SECONDARY_CONTROL") {
+				attributeState "battery", label: 'battery ${currentValue}%', unit: "%"
+			}
+		}
+		main "switch"
+		details(["switch"])
+	}
+}
+
+def getDOUBLE_STEP() {10}
+def getSTEP() {5}
+
+def parse(String description) {
+
+	def descMap = zigbee.parseDescriptionAsMap(description)
+
+	if (descMap && descMap.clusterInt == 0xFC10) {
+		def currentLevel = device.currentValue("level") as Integer ?: 0
+		def value = 0
+		if (descMap.data[0] == '01') {
+			//short press of 'ON' button
+			sendEvent(name: "switch", value: "on")
+		} else if (descMap.data[0] == '02' ) {
+			// move up
+			if (descMap.data[2] == '02') {
+				//long press of 'BRIGHTEN' button
+				value = Math.min(currentLevel + DOUBLE_STEP, 100)
+			} else if (descMap.data[2] == '01') {
+				//short press of 'BRIGHTEN' button
+				value = Math.min(currentLevel + STEP, 100)
+			}
+			sendEvent(name: "switch", value: "on")
+			sendEvent(name: "level", value: value)
+			
+		} else if (descMap.data[0] == "03") {
+			//move down
+			if (descMap.data[2] == '02') {
+				//long press of 'DIM' button
+				value = Math.max(currentLevel - DOUBLE_STEP, 0)
+			} else if (descMap.data[2] == '01') {
+				//short press of 'DIM' button
+				value = Math.max(currentLevel - STEP, 0)
+			}
+			
+			if (value == 0) {
+				sendEvent(name: "switch", value: "off")
+				sendEvent(name: "level", value: value)
+			} else {
+				sendEvent(name: "level", value: value)
+			}
+		} else if (descMap.data[0] == '04') {
+			//short press of 'OFF' button
+			sendEvent(name: "switch", value: "off")
+		} else if (descMap.data[0] == '06') {
+			//long press of 'ON' button
+			 sendEvent(name: "switch", value: "off")
+		} else if (descMap.data[0] == '08') {
+			//long press of 'OFF' button
+			sendEvent(name: "switch", value: "off")
+		}
+	} else if (descMap && descMap.clusterInt == 1 && descMap?.value) { 
+		def batteryValue = zigbee.convertHexToInt(descMap.value) / 2
+		sendEvent(name: "battery", value: batteryValue)
+	}
+}
+
+def off() {
+	sendEvent(name: "switch", value: "off", isStateChange: true)
+}
+
+def on() {
+	sendEvent(name: "switch", value: "on", isStateChange: true)
+}
+
+def setLevel(value, rate = null) {
+	if (value == 0) {
+		sendEvent(name: "switch", value: "off")
+	} else {
+		sendEvent(name: "switch", value: "on")
+		sendEvent(name: "level", value: value)
+	}
+}
+
+def installed() {
+	sendEvent(name: "switch", value: "on", displayed: false)
+	sendEvent(name: "level", value: 100, displayed: false)
+}
+
+def configure() {
+	zigbee.configureReporting(0x0001, 0x0021, DataType.UINT8, 30, 300, null) + zigbee.readAttribute(0x0001, 0x0021) 
```

Comment:
I assume you set the max report interval to 300 because you plan to add health check. If that's not the case then you can increase this significantly, to at least 6 hours.

Reply:
ok, I will extend time